Makefile

```diff
@@ -0,0 +1,172 @@
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCKER   ?= docker
+MKDIR    ?= mkdir
+TR       ?= tr
+DIST_DIR ?= $(CURDIR)/dist
+
+include $(CURDIR)/common.mk
+
+BUILDIMAGE_TAG ?= golang$(GOLANG_VERSION)
+BUILDIMAGE ?= $(IMAGE_NAME)-build:$(BUILDIMAGE_TAG)
+
+CMDS := $(patsubst ./cmd/%/,%,$(sort $(dir $(wildcard ./cmd/*/))))
+CMD_TARGETS := $(patsubst %,cmd-%, $(CMDS))
+
+CHECK_TARGETS := assert-fmt vet lint ineffassign misspell
+MAKE_TARGETS := binaries build check vendor fmt lint-internal test examples cmds coverage generate $(CHECK_TARGETS)
+
+TARGETS := $(MAKE_TARGETS) $(CMD_TARGETS)
+
+DOCKER_TARGETS := $(patsubst %,docker-%, $(TARGETS))
+.PHONY: $(TARGETS) $(DOCKER_TARGETS)
+
+GOOS ?= linux
+
+binaries: cmds
+ifneq ($(PREFIX),)
+cmd-%: COMMAND_BUILD_OPTIONS = -o $(PREFIX)/$(*)
+endif
+cmds: $(CMD_TARGETS)
+$(CMD_TARGETS): cmd-%:
+	CGO_LDFLAGS_ALLOW='-Wl,--unresolved-symbols=ignore-in-object-files' GOOS=$(GOOS) \
+		go build -ldflags "-s -w -X main.version=$(VERSION)" $(COMMAND_BUILD_OPTIONS) $(MODULE)/cmd/$(*)
+
+build:
+	GOOS=$(GOOS) go build ./...
+
+examples: $(EXAMPLE_TARGETS)
+$(EXAMPLE_TARGETS): example-%:
+	GOOS=$(GOOS) go build ./examples/$(*)
+
+all: check test build binary
+check: $(CHECK_TARGETS)
+
+# Update the vendor folder
+vendor:
+	go mod vendor
+
+# Apply go fmt to the codebase
+fmt:
+	go list -f '{{.Dir}}' $(MODULE)/... \
+		| xargs gofmt -s -l -w
+
+assert-fmt:
+	go list -f '{{.Dir}}' $(MODULE)/... \
+		| xargs gofmt -s -l > fmt.out
+	@if [ -s fmt.out ]; then \
+		echo "\nERROR: The following files are not formatted:\n"; \
+		cat fmt.out; \
+		rm fmt.out; \
+		exit 1; \
+	else \
+		rm fmt.out; \
+	fi
+
+ineffassign:
+	ineffassign $(MODULE)/...
+
+lint:
+# We use `go list -f '{{.Dir}}' $(MODULE)/...` to skip the `vendor` folder.
```

Comment:
Is that necessary?

I'm pretty sure `golangci-lint run ./...` will automatically ignore the vendor directory. Besides, isn't it empty now?

Reply:
It was necessary with golint, but maybe not for `golangci-lint`

Reply:
@pohly any ideas why it's killed in my env?
```
$ golangci-lint run -v ./...
INFO [config_reader] Config search paths: [./ /home/ed/git/dra-example-driver /home/ed/git /home/ed /home /] 
INFO [lintersdb] Active 7 linters: [errcheck gosimple govet ineffassign staticcheck typecheck unused] 
INFO [loader] Go packages loading at mode 575 (deps|files|imports|name|compiled_files|exports_file|types_sizes) took 1.101999526s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 7.466208ms 
Killed
```

Reply:
It gives more info after upgrade to the latest version:
```
$ golangci-lint run -v ...
INFO [config_reader] Config search paths: [./ /home/ed/git/dra-example-driver /home/ed/git /home/ed /home /] 
INFO [lintersdb] Active 7 linters: [errcheck gosimple govet ineffassign staticcheck typecheck unused] 
INFO [loader] Go packages loading at mode 575 (name|types_sizes|compiled_files|deps|exports_file|files|imports) took 3.446778534s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 652.361181ms 
INFO [linters_context/goanalysis] analyzers took 12m43.215635599s with top 10 stages: buildir: 3m2.48361962s, unused: 1m3.340601806s, directives: 22.824231005s, S1038: 20.786168519s, SA4006: 11.913960363s, SA4000: 9.809498718s, bools: 9.219121371s, isgenerated: 7.990880308s, errcheck: 7.556130456s, S1039: 7.493416405s 
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "github.com/kubernetes-sigs/dra-example-driver/pkg/example.com/resource/clientset/versioned/typed/nas/v1alpha1" 
INFO [runner] processing took 2.633µs with stages: skip_dirs: 354ns, max_same_issues: 343ns, nolint: 289ns, max_from_linter: 212ns, source_code: 143ns, identifier_marker: 133ns, autogenerated_exclude: 132ns, path_prettifier: 128ns, skip_files: 122ns, cgo: 114ns, filename_unadjuster: 114ns, fixer: 112ns, path_prefixer: 100ns, exclude: 55ns, sort_results: 47ns, path_shortener: 42ns, exclude-rules: 40ns, uniq_by_line: 39ns, diff: 38ns, max_per_file_from_linter: 38ns, severity-rules: 38ns 
INFO [runner] linters took 21.758707875s with stages: goanalysis_metalinter: 21.758670643s 
ERRO Running error: 1 error occurred:
	* can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "github.com/kubernetes-sigs/dra-example-driver/pkg/example.com/resource/clientset/versioned/typed/nas/v1alpha1"
 
INFO Memory: 257 samples, avg is 2938.4MB, max is 6014.6MB 
INFO Execution took 25.871098655s             
```

Reply:
Updated

Reply:
Regarding the vendor directory -- it's not commited back, but it's also not empty.

Reply:
@bart0sh I don't see any of these issues you are seeing. Neither on Mac or Linux. And neither native or in docker.

Reply:
It's definitely strange, but I've not seen anyone else report problems (albeit only 3 or 4 people have tried it out). It would be good to understand what the problem is if anyone else encounters it, but let's not block this PR on it.

Reply:
> any ideas why it's killed in my env?

Could it be the out-of-memory killer? Check your kernel log, if you run this on Linux.

Reply:
I can reproduce it. golangci-lint starts to swap quickly, then eventually gets killed. Odd.

```
23462.902388] oom-kill:constraint=CONSTRAINT_NONE,nodemask=(null),cpuset=/,mems_allowed=0,global_oom,task_memcg=/user.slice/user-1000.slice/session-59.scope,task=golangci-lint,pid=8437,uid=1000
[23462.903529] Out of memory: Killed process 8437 (golangci-lint) total-vm:101937572kB, anon-rss:58837380kB, file-rss:0kB, shmem-rss:0kB, UID:1000 pgtables:183440kB oom_score_adj:0
[23465.190427] oom_reaper: reaped process 8437 (golangci-lint), now anon-rss:0kB, file-rss:0kB, shmem-rss:0kB
```